service_configuration_lib/spark_config.py

```diff
@@ -135,6 +137,15 @@ def get_aws_credentials(
         with open(aws_credentials_json, 'r') as f:
             creds = json.load(f)
         return (creds.get('accessKeyId'), creds.get('secretAccessKey'), None)
+    elif assume_web_identity:
```

Comment:
AFAIK from the offline discussion with qlo, this needs to be used when credentials path is not given.

I think this can be confusing for Spark Users in debugging who are giving service credentials path and assume_aws_identity solution. 

I would suggest put the logic of at most one of the options is given and fail otherwise.